### PR TITLE
mosquitto: 1.4 -> 1.4.14

### PR DESCRIPTION
**Pull request by volth**
Security update, for multiple CVE (http://mosquitto.org/)

### CODE REVIEW

**Comment by Mic92** [comment]
Does it also require the libwebsocket version?

**Comment by volth** [comment]
yes, it does not compile with old websocket

**Comment by Mic92** [comment]
No other package depends on it. I will backport both.

**Comment by Mic92** [comment]
Mhm, I get an build error:

```
In file included from /nix/store/s96g9b8rl9ab13v1by7n661h0hzpbgsw-glibc-2.25-49-dev/include/unistd.h:25:0,
                 from mosquitto.c:22:
/nix/store/s96g9b8rl9ab13v1by7n661h0hzpbgsw-glibc-2.25-49-dev/include/features.h:180:3: warning: #warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE" [-Wcpp]
 # warning "_BSD_SOURCE and _SVID_SOURCE are deprecated, use _DEFAULT_SOURCE"
   ^~~~~~~
In file included from mosquitto.c:47:0:
/nix/store/4i3j0wmjmhqij6ics806bwnsqikbiinm-libwebsockets-2.3.0/include/libwebsockets.h:173:16: fatal error: uv.h: No such file or directory
 #include <uv.h>
                ^
compilation terminated.
make[1]: *** [Makefile:15: mosquitto.o] Error 1
make[1]: Leaving directory '/tmp/nix-build-mosquitto-1.4.14.drv-0/mosquitto-1.4.14/src'
make: *** [Makefile:21: mosquitto] Error 2
builder for ‘/nix/store/c952f041a1z2yy5mqmzcafaydck58xrq-mosquitto-1.4.14.drv’ failed with exit code 2
error: build of ‘/nix/store/c952f041a1z2yy5mqmzcafaydck58xrq-mosquitto-1.4.14.drv’ failed
/run/current-system/sw/bin/nix-shell: failed to build all dependencies
```

**Comment by Mic92** [comment]
Looks like it wants libuv as well.

**Comment by Mic92** [comment]
Thanks!

**Comment by volth** [comment]
> No other package depends on it.

Actually, ```collectd``` does.
But in nixpkgs, ```mosquitto```  is not a dependency, so collectd's configure do not find it and build without.

File: pkgs/development/libraries/libwebsockets/default.nix [hunk 1]
```diff
@@ -11,7 +11,8 @@ stdenv.mkDerivation rec {
     sha256 = "1hv2b5r6sg42xnqhm4ysjvyiz3cqpfmwaqm33vpbx0k7arj4ixvy";
   };
 
-  buildInputs = [ cmake openssl zlib libuv ];
+  buildInputs = [ cmake openssl zlib ];
+  propagatedBuildInputs = [ libuv ];
```
Comment (volth) [inline pkgs/development/libraries/libwebsockets/default.nix]: maybe ```openssl``` too

Comment (Mic92) [reply pkgs/development/libraries/libwebsockets/default.nix]: Is it required for compiling (in example used in public header)?

Comment (volth) [reply pkgs/development/libraries/libwebsockets/default.nix]: yes, but surrounded with #ifdefs 
https://github.com/warmcat/libwebsockets/blob/master/lib/libwebsockets.h#L195

Comment (volth) [reply pkgs/development/libraries/libwebsockets/default.nix]: ```libuv``` is not a strict dependency as well, a simple websocket servers could be created without linking with ```libuv```.

Comment (Mic92) [reply pkgs/development/libraries/libwebsockets/default.nix]: yeah, maybe I should revert this.

Comment (Mic92) [reply pkgs/development/libraries/libwebsockets/default.nix]: c1de354a3db238ddd1c125a77fd931566b2dda47

**Pull request closed without merging**